app/src/main/java/org/fossasia/openevent/app/data/auth/AuthServiceImpl.java

```diff
@@ -25,6 +29,9 @@
     private final Repository repository;
     private final AuthApi authApi;
 
+    @Inject
+    Context context;
```

Comment:
Are you sure about this ? Context in a non android class is usually not recommended although this one is the application context so no memory leak I guess.

Reply:
This is Android Context, so no memory leak, but this will not be used in AuthService, but in a separate model

We can use it to wrap APIs used in other models

Like Preference

https://github.com/fossasia/open-event-android/blob/development/app/src/main/java/org/fossasia/openevent/general/data/Preference.kt

Used here

https://github.com/fossasia/open-event-android/blob/development/app/src/main/java/org/fossasia/openevent/general/auth/AuthHolder.kt


Now I can use and mock preference outside of Android Classes, use it as an interface, change implementation and do as I please

Similar to what we are doing in this project in ContextUtils

Another example https://github.com/fossasia/open-event-android/blob/development/app/src/main/java/org/fossasia/openevent/general/data/Resource.kt